pip/commands/install.py

```diff
@@ -251,12 +251,12 @@ def run(self, options, args):
                     progress_bar=options.progress_bar,
                 )
 
-                self.populate_requirement_set(
-                    requirement_set, args, options, finder, session, self.name,
-                    wheel_cache
-                )
-
                 try:
+                    self.populate_requirement_set(
```

Comment:
Ah, Okie... Thanks @takluyver for the explanation!

I haven't really looked into this PR, I've just been skimming over it for the most part.